cobra/io/dict.py

```diff
@@ -121,7 +121,12 @@ def reaction_to_dict(reaction):
     new_reaction = OrderedDict()
     for key in _REQUIRED_REACTION_ATTRIBUTES:
         if key != "metabolites":
-            new_reaction[key] = _fix_type(getattr(reaction, key))
+            if key == "lower_bound" and reaction.lower_bound in {float('inf'), float('-inf'), float('nan')}:
+                new_reaction[key] = str(_fix_type(getattr(reaction, key)))
+            elif key == "upper_bound" and reaction.upper_bound in {float('inf'), float('-inf'), float('nan')}:
+                new_reaction[key] = str(_fix_type(getattr(reaction, key)))
+            else:
+                new_reaction[key] = _fix_type(getattr(reaction, key))
             continue
```

Comment:
The list `_REQUIRED_REACTION_ATTRIBUTES` is a constant list containing 6 entries at the moment. The loop iterates through all 6 elements but for 3 of them there are special cases. Depending on how often the function `reaction_to_dict(reaction)` is executed, this could have some run-time implications. It would be better not to use a loop at all but to cover the 6 cases directly.

Reply:
> It would be better not to use a loop at all but to cover the 6 cases directly.

 
Well, I can't say much about it. Each component of the COBRA model (reaction, metabolite, gene) has separate lists to define their required and optional attributes. That's how it was initially done, so doing it separately for the reaction would make it a little out of the order.

> Depending on how often the function reaction_to_dict(reaction) is executed, this could have some run-time implications.

That's surely a point to be noted. The extra comparisons introduced to handle `inf` values for lower or upper bound can be handled more efficiently if made outside the `for` loop. It will reduce the number of comparisons and will surely increase the efficiency.